### DISCUSSION TITLE
Item constructor doesn't make sense in my view.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#4392] [Ideas]
One of the overloads takes an ItemAsset just to convert it to an ID, the method that the ID is passed into just grabs the ItemAsset again. That doesn't really make sense. 
![image](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/67397422/a50bf299-083f-457d-bf63-ceb066191ace)

Expected setup:
```csharp
        public Item(ItemAsset asset, EItemOrigin origin)
        {
            if (itemAsset == null)
            {
                _id = 0;
                state = new byte[0];
                return;
            }

            _id = asset.id;

            if (origin == EItemOrigin.WORLD && !Provider.modeConfigData.Items.Has_Durability)
            {
                origin = EItemOrigin.CRAFT;
            }

            if (origin != 0)
            {
                amount = MathfEx.Max(itemAsset.amount, 1);
                quality = 100;
            }
            else
            {
                amount = MathfEx.Max(itemAsset.count, 1);
                quality = MathfEx.Clamp(itemAsset.quality, 0, 100);
            }

            state = itemAsset.getState(origin);
        }

        public Item(ushort newID, EItemOrigin origin) : this(((ItemAsset)Assets.find(EAssetType.ITEM, id)), origin), 
        {

        }
```

Luckily Assets.find it only grabs a key from a dictionary, but mind as well avoid calling it if you don't need to.

**Comment by SDGNelson** [comment]
As-is, yes, it doesn't really make sense - the item and inventory stuff would really benefit from some rewrites. I believe the intention there, however, was to avoid adding yet another place that creates an item by ID rather than asset reference. i.e., in the hypothetical rewrite that's one less place that needs to be adjusted.
